src/http-data-source.ts

```diff
@@ -390,15 +403,20 @@ export abstract class HTTPDataSource<TContext = any> extends DataSource {
             return cachedResponse
           }
           const response = this.performRequest<TResult>(options, cacheKey)
-          this.memoizedResults.set(cacheKey, response)
+          if (this.isRequestMemoizable(request)) {
```

Comment:
Do you mean encapsulating the memoization check in `performRequest`?

Reply:
Just noticing that the cachability check happens in `performRequest` method and wondering if it would make sense to implement in a similar fashion or if there is a difference that makes it necessary to implement differently.

Reply:
Even if the cache is set to `disable`, the response is memoized. 

Since the memoized verification is realized before the cachability check and before the request is perform, the memoized value is returned, independent if the request is set to `cacheable` or not.

Reply:
Understood. Definitely value in having a separate concept. My question is one of location of invocation. Checking for cacheability is a parallel concept, and that functionality is invoked within `performRequest` method. My question is if it would make sense to perform the memoizability check within the `performRequest` method as well. Somewhere like like 327.

Reply:
I think it makes sense to add the memoize verification inside the `performPequest` as well. But it will be necessary to change the `memoizedResults` type from `QuickLRU<string, Promise<Response<any>>>` to `QuickLRU<string, <Response<any>>`. I'll submit this modification.